.github/workflows/build.yml

```diff
@@ -22,3 +22,34 @@ jobs:
           cache: gradle
       - run: |
           ./gradlew spotlessCheck build publishToMavenLocal --no-daemon
+  reference-checker-tests:
+    if: github.base_ref == 'main' || github.base_ref == 'samples-google-prototype'
+    name: Run Reference Checker Tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the code
+        uses: actions/checkout@v3
+        with:
+          path: jspecify
+      - name: Check out the reference checker
+        uses: actions/checkout@v3
+        with:
+          repository: ${{ github.repository_owner }}/nullness-checker-for-checker-framework
+          path: nullness-checker-for-checker-framework
+      - name: Set up Java
+        uses: actions/setup-java@v3
+        with:
+          distribution: temurin
+          java-version: 17
+      - name: Run Conformance Tests
+        if: github.base_ref == 'main'
+        uses: gradle/gradle-build-action@v2
+        with:
+          arguments: ConformanceTest
+          build-root-directory: nullness-checker-for-checker-framework
+      - name: Run Samples Tests
```

Comment:
Should we make this bit and the conformance tests separate jobs so that we can run them in parallel?

Reply:
Which bit? The "samples" tests? We're only running the samples tests for PRs to the samples-google-prototype branch, and we're only running the conformance tests for PRs to the main branch.

Reply:
Yeah that's the one I meant--github's contextualization of comments is so-so. And ah, yeah the conditional takes care of that. Still, it might be more modular to factor these out to separate jobs—then again, we'd just be doing extra work.